packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -86,6 +88,27 @@ export class FargateService extends BaseService implements IFargateService {
       throw new Error('A TaskDefinition must have at least one essential container');
     }
   }
+
+  /**
+   * Adds AWS X-Ray daemon as a sidecar container to enable tracing
+   */
+  public addTracing(props: TracingOptions = {}) {
+    // TODO: adjust task size based on container-level resources (cpu/memory)?
+    const optIn = props.createLogs !== undefined ? props.createLogs : true;
+    const xray = this.taskDefinition.addContainer("xray", {
+      image: ContainerImage.fromRegistry("amazon/aws-xray-daemon"),
+      cpu: props.cpu,
+      memoryReservationMiB: props.memoryReservationMiB,
+      essential: props.essential || false,
+      logging: optIn ? this.createAWSLogDriver(this.node.id) : undefined,
+    });
+
+    xray.addPortMappings({
+      containerPort: 2000,
+      protocol: Protocol.Udp
+    });
```

Comment:
No links yet?

Reply:
For fargate there won't be links; only on EC2Service.